servicepulse/intro-failed-messages.md

```diff
@@ -65,37 +65,54 @@ The following actions can be performed on a failed message group:
  * **Archive group** - Archives all messages contained in the group. [Learn more about archiving messages](/servicepulse/intro-archived-messages.md).
 
 
-#### Viewing individual messages
-
-Individual messages views allow for accessing in-depth details about a given failed message, or retrying individual messages.
+### Listing messages
 
 Individual failed messages can viewed using one of the following two ways:
 
-- **Inside a failed message groups** - in the "Failed Messages Group" tab, click the "View messages" link from a failed message group entry
-- **All messages without any grouping** - in the "All messages" tab
+- **Inside a failed message group** - in the "Failed Messages Group" tab, click the "View messages" link from a failed message group entry
+- **All messages without any grouping** - via the "All messages" tab
 
 ![Failed Messages Page](images/intro-failed-messages-failed-messages-page.png 'width=500')
 
-Both of these individual message list views allow for taking actions on an individual message, on custom message selections or on all messages contained in the view.
+Both of these message list views allow for taking actions on an individual message, on custom message selections or on all messages contained in the view.
 
 NOTE: Retrying one or a few individual messages can be useful for testing system fixes before deciding to retry several messages in a group. This is because retrying several messages take a long time and queue other ServiceControl operations for longer than desired.
 
+The following actions can also be taken on each individual message or selection of messages:
+
+* **Retry** - Sends message(s) to be reprocessed by the corresponding endpoint.
+* **Archive** - Archives message(s).
+
+
+### Message details page
+
+As of ServicePulse v1.8.0, each individual message can be browsed in order to see in-depth details about a given failed message, archive or retry that message.
+
+![Failed Messages Page](images/failed-message-page.png 'width=500')
+
+Individual messages can be accessed by clicking the respective entry in any of the message list views.
+
+Each invidual failed message view allows for viewing:
+
+* **Message metadata** - Failure timestamp, endpoint name and location, retry status.
+* **StackTrace** - Full .NET exception stacktrace.
+* **Headers** - Complete set of message headers.
+* **Body** - Serialized message body.
+
+The following actions can also be taken on any given message:
 
-##### Managing individual failed messages
+* **Retry** - Sends message to be reprocessed by the corresponding endpoint.
```

Comment:
>Sends message to be reprocessed by the corresponding endpoint.

I am not sure if corresponding endpoint clearly conveys the meaning. I think it would be better as:

Sends message to be reprocessed by the same endpoint that failed to process it and caused the message to end up in the error queue.